controllers/component_build_controller.go

```diff
@@ -265,6 +261,27 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			// transient error, retry
 			return ctrl.Result{}, err
 		}
+
+		// when pipeline annotation was missing, we wil update component with default annotation
+		// so this reconcile will finish without error, and updated component will trigger new reconcile which already has pipeline annotation
+		// we don't want to update neither status or remove annotation, because that will be handled by next reconcile
+		missingPipelineAnnotationError := boerrors.NewBuildOpError(boerrors.EMissingPipelineAnnotation, nil)
+		if err.(*boerrors.BuildOpError).GetErrorId() == missingPipelineAnnotationError.GetErrorId() {
+			err = r.SetDefaultBuildPipelineComponentAnnotation(ctx, &component)
+
+			if err != nil {
+				if boErr, ok := err.(*boerrors.BuildOpError); ok && boErr.IsPersistent() {
+					buildStatus.Message = fmt.Sprintf("%d: %s", err.(*boerrors.BuildOpError).GetErrorId(), err.(*boerrors.BuildOpError).ShortError())
+				} else {
+					// transient error, retry
+					return ctrl.Result{}, err
+				}
+			} else {
+				return ctrl.Result{}, nil
```

Comment:
Having `if err == nil` { return ctrl.Result{}, nil }` before error handling will simplify code reading a bit, so we won't need else block at all.

Reply:
ok